src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,36 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, CompletableFuture<UserData>> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        // Use computeIfAbsent to initiate the data-fetching CompletableFuture only if absent
+        return cache.computeIfAbsent(userId, id -> fetchUserDataAsync(id));
```

Comment:
While using `computeIfAbsent` is thread-safe, be cautious with blocking operations inside the lambda expression, as they could potentially reduce the concurrency benefits. If `fetchUserDataAsync` is expected to be a long-running I/O operation, consider refactoring to avoid blocking the computation inside `computeIfAbsent`.